### DISCUSSION TITLE
Rotary Encoder not working.

### DISCUSSION THREAD

**Discussion by jhatchet** [ElectroTechnique/electrotechnique.github.io#22] [Hardware Issues]
I have a feeling I'm missing something in the setup, but I have everything working except the rotary encoder. It works to click it, but turning it does nothing at all. Any ideas?

**Comment by Electrotechnique** [comment]
What is the make and model of the encoder? Also the obvious, check soldering.

**Comment by yogiVan** [comment]
Hi thought I'd  reuse this post as  I have the same prob.
 I've built up two TSynths , they are v1.2 boards with the patches for v1.3. I had used some ebay encoders that I think are EC11 types, but after testing both boards and both having "non working stepping"  I ordered the exact PEC11Rs.  Removed one of the mystery encoders and replaced with a PEC11R. But still the same . the push switch works on all  but no scrolling.
I've continuity tested the encoder legs to the pins  (6 & 7) of the Teensy stack (4.1 with mounted Audio board) and tried reflowing the solder on Teensy pins. Center encoder leg connected to GND.
I will try uploading the latest fimware as I think I have v2.20.

**Comment by yogiVan** [comment]
So found the issue.  Seems the board v1.2 to v1.3  mod regarding the display Backlight pin connection to Teensy4.1 pin 6 was interfering with the encoder operation. 
Removed it and  BINGO full function \o/ So now will  revert the BLK pin to the original board connection but ATM its working fine being not connected to anything.

**Reply by Chamfer** [reply]
I'm having the same problem you describe. My board is the 1.3 revision, so I don't have a 'mod' to remove. From what you're saying, I think it means I could cut the trace marked in the photo for the same effect. Is that correct?

![backlight](https://user-images.githubusercontent.com/17420338/153693271-79a67c47-5216-4da2-b6dd-dff3530a4ff4.png)

**Comment by Electrotechnique** [comment]
Hello, no don't cut that trace it's the LCD backlight. What encoder are you using?

**Reply by Chamfer** [reply]
I'm glad I asked!
I don't know specifically (I can find out, but it will require some disassembly - tomorrow). The kit is from Analog Lab Swiss (https://www.etsy.com/listing/964949811/diy-12-notes-polyphonic-electrotechnique)  and my impression of it is that everything is good quality. I doubt he would have included a garbage encoder.

It is of course quite possible that I cooked it while soldering. I don't think that's the case, but I cannot rule it out. I have reflowed the solder points, and also gave it a few squirts of an electronics contact cleaner. No success.

The push-button click portion of the encoder works fine.

**Reply by Electrotechnique** [reply]
Sorry to say this but just about all problems so far are soldering problems. If you have multimeter, you could try looking at the voltage across the encoder A and B pins, as it's turned and see if you can detect it dipping.

Yes, the parts supplied by Analog Lab Swiss are correct.

**Reply by yogiVan** [reply]
Well on my boards the displays were pulling the BL line (teensy pin 6) high. Their internal pull-ups are strong enough to over ride the encoder pin.  They function just fine with BL disconnected/floating.
I can't speak to every situation just what work for me. On my modded 1.2 boards I just cut the 'blue' wires and the encoders worked. My best guess is not all displays have the same internal pull-up values.

**Reply by Electrotechnique** [reply]
The displays have BL pulled up on them, so no connection is required. The suggestion of modding 1.2 boards with a connection to the BL is not essential. I considered that it might be useful to have control of the backlight in future if there was work on a soft on/off function (TSynth has no/off switch), which requires turning off the backlight.

Really don't touch the BL pin, you don't need to. The BL pin has not effect on the encoder. 

I suspect you have connected the BL pin wrongly when modding it. Look again at https://github.com/ElectroTechnique/electrotechnique.github.io/discussions/6   note 2.

**You connect BL to the pin marked 8 on the picture, which is CALLED pin 6. I think you connected it to the pin marked 6, which is Encoder A.**

![Screenshot 2022-02-13 115213](https://user-images.githubusercontent.com/50934180/153736184-19d8e6fb-cda0-444c-ae9a-0f1454576b98.jpg)

**Reply by Chamfer** [reply]
I have resoldered the display pins, the encoder pins, all the Teensy pins (both Teensy/connectors and connectors/mainboard), and a number of the potentiometers as well.
If anything, I've made it worse. Part way through the process when I tested, any flex to the board would cause the display to cycle through various random seeming screens. Any by 'flex' I don't mean I was grabbing the corners and twisting - I mean the board was on its side, and I was tilting it different ways with a single finger. This *definitely* would seem to support your suggestion of a solder problem.
Now, the display won't light at all. When I plug in power, it 'darkens', but never shows any data. If I disconnect the Teensy assembly from the mainboard and try it with USB and headphones, I get no sound except some random faint clicks and pops. 

At this point, I suspect I've actually damaged the Teensy - either the silicon, or the USB connection (which definitely seems loose).

I'll be ordering another Teensy and trying again. Could anyone suggest what size/spec of pogo pins to use to connect the D+ and D- pads for the USB clients (per 5. Attaching the Teensy)?

I appreciate the help - and the patience :-)

**Reply by Electrotechnique** [reply]
I think a high resolution picture of both sides of the PCB would be useful.

The display rapidly cycling through parameter settings means that the Teensy isn't receiving signals from the two MUXs properly and random noise on the ADCs are causing parameter changes. With the Teensy off the PCB try grounding the MUX  pins and test. Use a MIDI program like MIDI Ox to test the Teensy using the USB as MIDI in.

![Screenshot 2022-02-13 220151](https://user-images.githubusercontent.com/50934180/153754422-a034f06c-8a41-4b85-be6b-d7fe5a05fff9.jpg)

**Comment by Chamfer** [comment]
Sorry about the delay replying. I was having trouble getting the hi-res images. They keep coming out hi-blur.

Anyhow, I believe I had identified the problem. It is soldering related, as you suggested, just not in the way I expected. Tonight I plugged the Teensy/Audio board assembly into the mainboard and started measuring continuity of those pins and sockets. So for example I measured from the top of the Teensy at pin 1 to where the corresponding solder point was on the bottom of the main board. Most of them were fine, but a whole swath, primarily the pins adjacent to the headphone jack on the audio board, were not. 

In other words, the Teensy/Audio board assembly is not seating properly in its socket strips. I suspect this is caused by solder that creeped up the pins on the Teensy side, causing them to not quite fit into their sockets. And if I get one side seated as well as possible, the display comes on, but the whole other side wants to come up away from its socket a mm or so - the problem just moves around.

Also, I hooked the whole assembly up to a PC. Using VMDK I was able to send it notes, which it played just fine, and send program change messages to select different sounds from the microSD. Though at this point the display was blank, I tested all the potentiometers and almost all of them had clean, non-random effects on the sound. Which tells me the MUXs are almost certainly working just fine.

Even if my holding my mouth just right I was able to get the Teensy seated in the socket such that everything worked, I could never trust it, knowing that the slightest jarring would knock it into uselessness again. So it seems to me that the best course of action would be to remove the header sockets from the main board, and solder the Teensy/Audio board directly to it. That eliminates the failure zone entirely.

I'm going to have to research best practices for desoldering something like these 24-point headers. Desolder pump and wicks? I'm twitchy just thinking about it.

**Reply by Chamfer** [reply]
And finally a followup. I got a fancy desoldering tool today, and - with some fiddling and muttering - managed to remove the headers from the main board. Soldered the Teensy/Audio assembly directly to the board, and it's working perfectly. Display, encoder, all the pots. 

Thank you for stepping in with suggestions. It helped a lot to know I wasn't doing this in a vacuum.
